projects/storefrontapp/tsconfig.app.json

```diff
@@ -4,8 +4,10 @@
     "module": "es2020",
     "outDir": "../../out-tsc/app",
     "strict": false, // TODO: Fix strict errors and remove this option
-    "types": [],
-    "target": "es2015"
+    "types": ["@sapui5/ts-types-esm"],
+    "target": "es2015",
+    "allowUmdGlobalAccess": true,
+    "skipLibCheck": true
```

Comment:
1. Are those 2 tsconfig flags really needed? Why we need them?
2. Do customers need to enable those 2 flags too in their codebase? If yes, when?

PS. Besides explanation here, it would be good to also drop a comment in the file, for reference

Reply:
`skipLibCheck` is needed to compile the code and yes customers will need this too. This is due to issues in the UI5 type definitions.

I will check whether the `allowUmdGlobalAccess` configuration can be removed.

Reply:
I'm wondering how much the option `skipLibCheck` degrades the type-safety for our internal development in our monorepo and also in customers' codebases...

I'd love to avoid ti find a different workaround, if possible. Or at least scope this tsconfig only inside your library's tsconfig, if possible.

PS. If customers _really_ must use it (which I still hope they don't), then we would also need to add to our Spartacus installation schematics a script that adds this config option in customer's tsconfig...

Reply:
I have read that `skipLibCheck` is not meant to be great in a monorepo, but I have not hit any issues with it at all so far. Type safety has been great. However, I understand that you would rather not force this on customers.

I have added code in the schematic for the epd-visualization library to set `skipLibCheck: true`, however this code would need to go in some other schematic that always runs given the dependencies on @sapui5/ts-types-esm.  

I have seen a hack that actually modifies the extracted .d.ts files in the node_modules directory and inserts a `// @ts-nocheck` comment. This might avoid the need to use `skipLibCheck`, but it feels a bit dirty.

I will experiment with package dependencies to see whether there is any other option.

Reply:
I haven't found any solution that I think is better than just using `skipLibCheck`.

Modifying the `node_modules/@sapui5/ts-types-esm/types/*.d.ts` files to include `// @ts-nocheck` does suppress the compilation errors. Making this work for customers would involve using a schematic to modify the customer's build system to include the hack . This seems worse to me than the `skipLibCheck` option. I haven't got a working solution for this, but I think it could work.

I didn't find any other approach that worked.

I spent a fair amount of time trying to find a good solution to this before this review, but I haven't found any new solutions this time around.

What are your thoughts on how to proceed?

Reply:
I need to think about it

1. Regarding our monorepo, I'm still not convinced that one feature (integration) lib should degrade the type safety of the whole project.
Did you consider having a separate tsconfig with `skipLibCheck`, e.g. tsconfig.app-epd.json that could be used only for builds when EPD feature is enabled?

2. Regarding customer's app, do they really need the config `skipLibCheck` in case they're just _using_ the Spartacus EPD integration, but they're _not extending_ our visualisation services (i.e. _not refering directly_ to any ui5 types in their codebase)?

Reply:
Ad 1.
Only now I realized that `skipLibCheck` in tsconfig.app.json is not that bad as I initially thought. 
In particular, the builds of all individual libraries (that we ship to npm) are not affected. They are built using individual tsconfigs that don't have `skipLibCheck`. The `tsconfig.app.json` affects just the build of our test shell app, but not the whole monorepo.
(TBH I was mostly afraid that after your change we would be skipping checks of d.ts of our library A when compiling our another library B that depends on A. But fortunately it's not the case!)

**I'm no longer so much concerned about `skipLibCheck` in our app.tsconfig.json. And I'm sorry for confusion.**

Reply:
Ad 2.
Answer: Yes, customers need to have the `skipLibCheck` enabled, even when EPD feature lib is just used OOTB. 

I've checked it in a fresh Angular app + Spartacus installed via schematics, including the installation of @spartacus/epd-visualisation

It's becasue the build of the customer's app lookups transitively any _used_ types (in d.ts)
```
epd-visualisation-feature.module.ts (in customer's app, referencing `EpdVisualisationModule`) 
 -> epd-visualisation.module.d.ts (in node_modules/@spartacus/epd-visualisation/...)
 -> ... 
 -> visual-viewer.service.d.ts (in node_modules/@spartacus/epd-visualisation/...)
 -> BROKEN TYPES in node_modules/@sapui5/ts-types-esm/types/...
```

Reply:
Ad 2. 
Update: There might be a chance might avoid `skipLibCheck` in customer's app. See my another comment https://github.com/SAP/spartacus/pull/14420#discussion_r769830695 on removing the property `protected _ui5: ui5;`

Reply:
I have implemented your solution using `/// <reference types="@sapui5/ts-types-esm" />`